pkg/hash/pool.go

```diff
@@ -0,0 +1,32 @@
+package hash
+
+import (
+	"fmt"
+	"sync"
+)
+
+// NewPool returns a sync.Pool of a Hasher with the given hashName
+// It creates one hash to check the hash name is supported
+// (which is then put in the pool), to avoid panic()'ing later.
```

Comment:
```suggestion
// NewPool returns a sync.Pool of a Hash with the given hashType
// It creates one Instance to check the hash type is supported
// (which is then put in the pool), to avoid panic()'ing later,
// as the sync.Pool New() function doesn't allow errors.
```